### PR TITLE
Added `AcceptResponse` to allow automated mapping of the 202 status with the Location Header

### PR DESCRIPTION
**Pull request by oskardudycz**
- Added `AcceptResponse` to allow automated mapping of the 202 status with the Location Header
- Made `WolverineProducesResponseTypeMetadata` to allow providing custom `IHttpAware` implementation as described in docs,
- Updated C# language to the latest 12 version,
- Added docs and tests for AcceptResponse,
- Applied minor tweaks with the Boy Scout rule.

### CODE REVIEW

File: src/Http/Wolverine.Http/Resources/WolverineProducesResponseTypeMetadata.cs [hunk 1]
```diff
@@ -1,11 +1,8 @@
-using System;
-using System.Collections.Generic;
-using System.Diagnostics;
 using Microsoft.AspNetCore.Http.Metadata;
 
 namespace Wolverine.Http.Resources;
 
-internal class WolverineProducesResponseTypeMetadata : IProducesResponseTypeMetadata
+public class WolverineProducesResponseTypeMetadata : IProducesResponseTypeMetadata
```
Comment (oskardudycz) [inline src/Http/Wolverine.Http/Resources/WolverineProducesResponseTypeMetadata.cs:5]: **COMMENT:** Without that one wouldn't be able to create its own `IHttpAware` implementation.

File: Directory.Build.props [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project>
     <PropertyGroup>
-        <LangVersion>11</LangVersion>
+        <LangVersion>12</LangVersion>
```
Comment (oskardudycz) [inline Directory.Build.props:4]: **COMMENT:** Boy Scout rule 🫡

File: src/Http/Wolverine.Http/IHttpAware.cs [hunk 3]
```diff
@@ -108,4 +108,37 @@ void IHttpAware.Apply(HttpContext context)
...
+#region sample_AcceptResponse
+
+/// <summary>
+/// Base class for resource types that denote some kind of request being accepted in the system.
+/// </summary>
+public record AcceptResponse(string Url) : IHttpAware
+{
+    public static void PopulateMetadata(MethodInfo method, EndpointBuilder builder)
+    {
+        builder.RemoveStatusCodeResponse(202);
```
Comment (haefele) [inline src/Http/Wolverine.Http/IHttpAware.cs:123]: Is this supposed to remove 202?

`CreationResponse` removes 200, as that's one of the default status-codes wolverine adds.

File: src/Http/Wolverine.Http.Tests/Marten/using_aggregate_handler_workflow.cs [hunk 4]
```diff
@@ -165,20 +181,47 @@ public async Task use_stream_collision_policy()
...
+        });
+
+        var status = result.ReadAsJson<OrderStatus>();
+        status.ShouldNotBeNull();
+
+        result = await Scenario(x =>
+        {
+            x.Post.Json(new ConfirmOrder(status.OrderId)).ToUrl($"/orders/{status.OrderId}/confirm");
+            
+            x.StatusCodeShouldBe(204);
```
[the remark this answers is not in the document]
Comment (oskardudycz) [reply src/Http/Wolverine.Http.Tests/Marten/using_aggregate_handler_workflow.cs]: Nice catch, @haefele, my bad! Fixed in https://github.com/JasperFx/wolverine/pull/726/commits/41a746f55177f99015e502d0ec3f5ddc0a8c1167.

**Comment by oskardudycz** [comment]
@jeremydmiller, is it expected that no tests are run in the pipeline, `ci` target runs only compilation. See:
- https://github.com/JasperFx/wolverine/blob/main/build/build.cs#L92
- https://github.com/JasperFx/wolverine/blob/23d33125d9cd1ec921a5a2d25efe6c03e6137afe/.github/workflows/dotnet.yml#L44

**Pull request merged**